pkg/cloud/vsphere/services/metadata/metadata.go

```diff
@@ -29,6 +29,24 @@ import (
 // New returns the cloud-init metadata as a base-64 encoded string for a given
 // machine context.
 func New(ctx *context.MachineContext) ([]byte, error) {
+	net := ctx.MachineConfig.MachineSpec.Network
+
+	var format string
+
+	// skip network customization for basic DHCP networks reducing the risk of
```

Comment:
Hi @moshloop,

Hmm. I see. Good catch, but please yank this out of this PR and let's discuss it in a new PR. Please file an issue as well. This is a really good point -- I just don't want to tie it to the ESX work.

Reply:
So this is actually a dependency on getting ESXi to work, I don't think ESXi generates the mac address on create, but rather on power on, which is too late the metadata script.